scylla-cql/src/frame/frame_errors.rs

```diff
@@ -40,10 +43,10 @@ pub enum FrameError {
 
 #[derive(Error, Debug)]
 pub enum ParseError {
+    #[error("Failed to serialize EXECUTE request: {0}")]
+    ExecuteSerialization(#[from] ExecuteSerializationError),
     #[error("Failed to serialize QUERY request: {0}")]
     QuerySerialization(#[from] QuerySerializationError),
-    #[error("Failed to serialize query parameters: {0}")]
-    QueryParametersSerialization(#[from] QueryParametersSerializationError),
     #[error("Low-level deserialization failed: {0}")]
     LowLevelDeserializationError(#[from] LowLevelDeserializationError),
     #[error("Could not serialize frame: {0}")]
```

Comment:
> I assume it was introduced in first commit just to make commits atomic?

Exactly. I firstly introduced `QueryParametersSerializationError` - since I changed the return type of `QueryParameters::serialize`, I needed to add it to `ParseError` (returned by `SerializableRequest::serialize`). Then, I introduced two error types that depend on `QPSerializationError`, and thanks to that I could finally remove it from `ParseError`.